samples/excel/82-document/get-file-in-slices-async.yaml

```diff
@@ -0,0 +1,195 @@
+
+                OfficeHelpers.UI.notify("The base64-encoded string that represents the current document has been written to the text box below. To validate the string, use an online tool such as https://www.base64decode.org/ to decode the string, and then open the resulting file.");
+            }
+        }
+
+        function getFileContents(
+            file: Office.File,
+            onSuccess: (byteArray: number[]) => void,
+            onError: (error: Office.Error) => void
+        ) {
+            let expectedSliceCount = file.sliceCount;
+            let fileSlices: Array<Array<number>> = [];
+
+            console.log("Current file size in bytes: " + file.size);
+            console.log("Number of file slices: " + file.sliceCount);
+
+            getFileContentsHelper();
+
+            function getFileContentsHelper() {
+                file.getSliceAsync(fileSlices.length, function (result) {
+                    if (result.status === Office.AsyncResultStatus.Failed) {
+                        file.closeAsync();
+                        return onError(result.error);
+                    }
+
+                    // Got one slice, store it in a temporary array.
+                    fileSlices.push(result.value.data);
+
+                    if (fileSlices.length == expectedSliceCount) {
+                        console.log("All slices have been received.");
+                        file.closeAsync();
+
+                        let array = [];
+                        fileSlices.forEach(slice => {
+                            array = array.concat(slice);
+                        });
+
+                        onSuccess(array);
+                    } else {
+                        getFileContentsHelper();
+                    }
+                });
+            }
+        }
+
+        async function setup() {
+            await Excel.run(async (context) => {
+                const sheet = await OfficeHelpers.ExcelUtilities
+                    .forceCreateSheet(context.workbook, "Sample");
+                let expensesTable = sheet.tables.add('A1:E1', true);
+                expensesTable.name = "SalesTable";
+
+                expensesTable.getHeaderRowRange().values = [["Product", "Qtr1", "Qtr2", "Qtr3", "Qtr4"]];
+
+                expensesTable.rows.add(null, [
+                    ["Frames", 5000, 7000, 6544, 4377],
+                    ["Saddles", 400, 323, 276, 651],
+                    ["Brake levers", 12000, 8766, 8456, 9812],
+                    ["Chains", 1550, 1088, 692, 853],
+                    ["Mirrors", 225, 600, 923, 544],
+                    ["Spokes", 6005, 7634, 4589, 8765]
+                ]);
+
+                if (Office.context.requirements.isSetSupported("ExcelApi", 1.7)) {
+                    sheet.getUsedRange().format.autofitColumns();
+                    sheet.getUsedRange().format.autofitRows();
+                }
+
+                sheet.activate();
```

Comment:
I would move sheet.activate down by one (to only activate after chart creation -- I.e., as the very final call before the sync

Reply:
Incorped, thanks!